builtin/providers/aws/resource_aws_elasticache_replication_group.go

```diff
@@ -0,0 +1,231 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/awslabs/aws-sdk-go/aws"
+	"github.com/awslabs/aws-sdk-go/service/elasticache"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsElasticacheReplicationGroup() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsElasticacheReplicationGroupCreate,
+		Read:   resourceAwsElasticacheReplicationGroupRead,
+		Update: resourceAwsElasticacheReplicationGroupUpdate,
+		Delete: resourceAwsElasticacheReplicationGroupDelete,
+
+		Schema: map[string]*schema.Schema{
+			"replication_group_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"description": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"cache_node_type": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"automatic_failover": &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+			"num_cache_clusters": &schema.Schema{
+				Type:     schema.TypeInt,
+				Optional: true,
+				Default:  1,
+			},
+			"parameter_group_name": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"engine": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Default:  "redis",
+			},
+			"engine_version": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func resourceAwsElasticacheReplicationGroupCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).elasticacheconn
+
+	var hasParameterGroup bool
+
+	replicationGroupId := d.Get("replication_group_id").(string)
+	description := d.Get("description").(string)
+	cacheNodeType := d.Get("cache_node_type").(string)
+	automaticFailover := d.Get("automatic_failover").(bool)
+	numCacheClusters := d.Get("num_cache_clusters").(int)
+	parameterGroupName, hasParameterGroup := d.GetOk("parameter_group_name")
+	engine := d.Get("engine").(string)
+	engineVersion := d.Get("engine_version").(string)
+
+	req := &elasticache.CreateReplicationGroupInput{
+		ReplicationGroupID:		aws.String(replicationGroupId),
+		ReplicationGroupDescription:	aws.String(description),
+		CacheNodeType:            	aws.String(cacheNodeType),
+		AutomaticFailoverEnabled: 	aws.Boolean(automaticFailover),
+		NumCacheClusters:         	aws.Long(int64(numCacheClusters)),
+		Engine:				aws.String(engine),
+		EngineVersion:			aws.String(engineVersion),
+	}
+
+	if hasParameterGroup {
+		req.CacheParameterGroupName = aws.String(parameterGroupName.(string))
+	}
```

Comment:
nitpick: a pattern we've adopted here is to chick this `GetOk` in a block, like so:

```go
if v, ok := d.GetOk("parameter_group_name"); ok {
  req.CacheParameterGroupName = aws.String(v.(string))
}
```

This way you don't need a separate `parameterGroupName ` or `hasParameterGroup` variable, since neither are used after this step.